File: hackerrank/easy/breaking-records/breaking-records.swift
// Time Complexity: O(n)
// Space COmplexity: O(1)
func breakingRecords(scores: [Int]) -> [Int] {
    var minRecord = 0
    var maxRecord = 0
    
    var lowest = scores[0]
    var highest = scores[0]
    
    for score in scores {
      if score < lowest {
        minRecord += 1
        lowest = score
      } else if score > highest {
        maxRecord += 1
        highest = score
      }
    }
    
    return [maxRecord, minRecord]
}

File: hackerrank/easy/breaking-records/breaking-records.dart
// Time Complexity: O(n)
// Space Complexity: O(1)
List<int> breakingRecords(List<int> scores) {
  int minRecord = 0;
  int maxRecord = 0;

  int lowest = scores[0];
  int highest = scores[0];

  for (int score in scores) {
    if (score < lowest) {
      minRecord++;
      lowest = score;
    } else if (score > highest) {
      maxRecord++;
      highest = score;
    }
  }

  return [maxRecord, minRecord];
}
